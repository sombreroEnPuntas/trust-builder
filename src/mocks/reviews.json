[{
        "firstName": "Simon",
        "lastName": "Lock",
        "fullName": "Simon Lock",
        "location": "Kolding",
        "reviewTitle": "Super quality.. I will show here again!",
        "reviewBody": "Super nice quality, fast devilery, good prices. I will shop here again!",
        "starRating": "5"
    },
    {
        "firstName": "Gav",
        "lastName": "",
        "fullName": "Gav",
        "location": "",
        "reviewTitle": "Princely Sum",
        "reviewBody": "A decent local curry house in Faversham, Kent known for its Elvis nights.",
        "starRating": "4"
    },
    {
        "firstName": "Justin",
        "lastName": "Wright",
        "fullName": "Justin Wright",
        "location": "London, GB",
        "reviewTitle": "Good Services",
        "reviewBody": "A decent place to introduce your taste buds to fiery Indian fare",
        "starRating": "3"
    },
    {
        "firstName": "Erika",
        "lastName": "Wolfe",
        "fullName": "Erika Wolfe",
        "location": "Gothenburg, SE",
        "reviewTitle": "Nightmare experience - no product, no communication, no refund; improved by rapid resolution",
        "reviewBody": "In early 2012, I ordered a set of chairs from Infurn. I thought that by ordering in March, I would have what I needed by November, certainly. I wanted the perfect chairs for my house, and really did not NEED them before the annual Thanksgiving dinner (the only time of year I have a need for a whole lot of chairs at one time). I played it safe, I thought, by ordering so far in advance.\nWeek after week,\nmonth after month,\nInfurn 's website kept updating the status of the order so that the chairs'\narrival would be further and further in the distant future.Finally when the week of Thanksgiving arrived,\nand I had had contact with their customer service in early November(and their only clueless reply was to say, 'We had manufacturing problems; let'\ns hope the chairs arrive this week as scheduled.\n'), I still had no chairs, had to go out and buy some other chairs and still had no clue when the Infurn chairs might arrive.\nAt some point I\nfinally just requested a refund because Infurn could neither deliver my chairs nor give me a solid date about when I might receive them when I inquired about a delivery date.They\nfinally offered me a refund - which I accepted on 14 December 2012. ",
        "starRating": "2"
    },
    {
        "firstName": "Hugo",
        "lastName": "Beja",
        "fullName": "Hugo Beja",
        "location": "Praia Da Barra, PT",
        "reviewTitle": "FRAUD",
        "reviewBody": "I've been patiently waiting for a miracle to happen with our order AU-316084, 12 Chairs!! First the delays, lots and lots of delays with no apparent reason... after they send just 1/2 of the order and say it's all... So we bought 'Pairs' of chairs... we paid for 5 PAIRS and received 5 chairs!!! The other 2, the rocket chairs we never receive....\nCommunication ZERO,\nthey simply ignore the e - mails...when they did respond they asked us to prove our order to be pairs!!!LOL and just stopped communicating...their website is constantly down...probably to make lose interest and rest your forces to recover what you paid\nfor !!",
        "starRating": "1"
    },
    {
        "firstName": "Simona",
        "lastName": "Locka",
        "fullName": "Simona Locka",
        "location": "Brandenburg",
        "reviewTitle": "Ich hat eine ente",
        "reviewBody": "Die ente frist das hund",
        "starRating": "5"
    },
    {
        "firstName": "Gava",
        "lastName": "",
        "fullName": "Gava",
        "location": "",
        "reviewTitle": "Cesaro Summability",
        "reviewBody": "This was a song I was listening to when I was younger.",
        "starRating": "4"
    },
    {
        "firstName": "Justina",
        "lastName": "Wrighta",
        "fullName": "Justina Wrighta",
        "location": "Leeds, GB",
        "reviewTitle": "Good Lord",
        "reviewBody": "Train be late, ain't no waiting!",
        "starRating": "3"
    },
    {
        "firstName": "Eriko",
        "lastName": "Wolfo",
        "fullName": "Eriko Wolfo",
        "location": "Johanesburg",
        "reviewTitle": "Sunmare experience - yes product, yes communication, yes refund; improved by slow resolution",
        "reviewBody": "In early 2102, I ordered a set of stairs from Infurn. I thought that by ordering in March, I would have what I needed by November, certainly. I wanted the perfect stairs for my house, and really did not NEED them before the annual Thanksgiving dinner (the only time of year I have a need for a whole lot of stairs at one time). I played it safe, I thought, by ordering so far in advance.\nWeek after week,\nmonth after month,\nInfurn 's website kept updating the status of the order so that the stairs'\narrival would be further and further in the distant future.Finally when the week of Thanksgiving arrived,\nand I had had contact with their customer service in early November(and their only clueless reply was to say, 'We had manufacturing problems; let'\ns hope the stairs arrive this week as scheduled.\n'), I still had no stairs, had to go out and buy some other stairs and still had no clue when the Infurn stairs might arrive.\nAt some point I\nfinally just requested a refund because Infurn could neither deliver my stairs nor give me a solid date about when I might receive them when I inquired about a delivery date.They\nfinally offered me a refund - which I accepted on 14 December 2012.",
        "starRating": "2"
    },
    {
        "firstName": "Huga",
        "lastName": "Bejo",
        "fullName": "Huga Bejo",
        "location": "Fortaleza, BZ",
        "reviewTitle": "FRAUD",
        "reviewBody": "I've been patiently waiting for a miracle to happen with our order AU-316084, 12 stairs!! First the delays, lots and lots of delays with no apparent reason... after they send just 1/2 of the order and say it's all... So we bought 'Pairs' of stairs... we paid for 5 PAIRS and received 5 stairs!!! The other 2, the rocket stairs we never receive....\nCommunication ZERO,\nthey simply ignore the e - mails...when they did respond they asked us to prove our order to be pairs!!!LOL and just stopped communicating...their website is constantly down...probably to make lose interest and rest your forces to recover what you paid\nfor !!",
        "starRating": "1"
    },
    {
        "firstName": "Simon",
        "lastName": "Lock",
        "fullName": "Simon Lock",
        "location": "Kolding",
        "reviewTitle": "Super quality.. I will show here again!",
        "reviewBody": "Super nice quality, fast devilery, good prices. I will shop here again!",
        "starRating": "5"
    },
    {
        "firstName": "Gav",
        "lastName": "",
        "fullName": "Gav",
        "location": "",
        "reviewTitle": "Princely Sum",
        "reviewBody": "A decent local curry house in Faversham, Kent known for its Elvis nights.",
        "starRating": "4"
    },
    {
        "firstName": "Justin",
        "lastName": "Wright",
        "fullName": "Justin Wright",
        "location": "London, GB",
        "reviewTitle": "Good Services",
        "reviewBody": "A decent place to introduce your taste buds to fiery Indian fare",
        "starRating": "3"
    },
    {
        "firstName": "Erika",
        "lastName": "Wolfe",
        "fullName": "Erika Wolfe",
        "location": "Gothenburg, SE",
        "reviewTitle": "Nightmare experience - no product, no communication, no refund; improved by rapid resolution",
        "reviewBody": "In early 2012, I ordered a set of chairs from Infurn. I thought that by ordering in March, I would have what I needed by November, certainly. I wanted the perfect chairs for my house, and really did not NEED them before the annual Thanksgiving dinner (the only time of year I have a need for a whole lot of chairs at one time). I played it safe, I thought, by ordering so far in advance.\nWeek after week,\nmonth after month,\nInfurn 's website kept updating the status of the order so that the chairs'\narrival would be further and further in the distant future.Finally when the week of Thanksgiving arrived,\nand I had had contact with their customer service in early November(and their only clueless reply was to say, 'We had manufacturing problems; let'\ns hope the chairs arrive this week as scheduled.\n'), I still had no chairs, had to go out and buy some other chairs and still had no clue when the Infurn chairs might arrive.\nAt some point I\nfinally just requested a refund because Infurn could neither deliver my chairs nor give me a solid date about when I might receive them when I inquired about a delivery date.They\nfinally offered me a refund - which I accepted on 14 December 2012. ",
        "starRating": "2"
    },
    {
        "firstName": "Hugo",
        "lastName": "Beja",
        "fullName": "Hugo Beja",
        "location": "Praia Da Barra, PT",
        "reviewTitle": "FRAUD",
        "reviewBody": "I've been patiently waiting for a miracle to happen with our order AU-316084, 12 Chairs!! First the delays, lots and lots of delays with no apparent reason... after they send just 1/2 of the order and say it's all... So we bought 'Pairs' of chairs... we paid for 5 PAIRS and received 5 chairs!!! The other 2, the rocket chairs we never receive....\nCommunication ZERO,\nthey simply ignore the e - mails...when they did respond they asked us to prove our order to be pairs!!!LOL and just stopped communicating...their website is constantly down...probably to make lose interest and rest your forces to recover what you paid\nfor !!",
        "starRating": "1"
    }
]   