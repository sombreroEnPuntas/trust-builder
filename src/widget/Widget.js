import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Comment from './comment';
import { StarFill } from './star';
import withSpinner from './withSpinner';

import './Widget.scss';

const accumulateReviewRatings = (sum, review) => sum + parseInt(review.starRating, 10);
const averageReviewRating = reviews => reviews.reduce(accumulateReviewRatings, 0) / reviews.length;
const logoSrc = `${process.env.PUBLIC_URL}/logo/255x29.png`;

export class Widget extends Component {
  render() {
    const {
      reviews,
      currentCard,
      totalCards,
      onClickBack,
      onClickNext,
    } = this.props;

    if(reviews.length < 1) {
      return (<div>Loading reviews...</div>);
    }

    const cards = [];
    let index = 0;
    let cardIndex;

    for (; index <= 4; index++) {
      cardIndex = currentCard + index;
      cardIndex = cardIndex > totalCards - 1 ? cardIndex - totalCards : cardIndex;

      cards.push(
        <span key={`card-${index}`} className="positioning">
          <Comment review={reviews[cardIndex]} />
        </span>
      );
    }

    return (
      <div className="Widget">
        <button className="button button_type-back" onClick={onClickBack}>&lt;</button>
        <div className="comments">
          {cards}
        </div>
        <button className="button button_type-next" onClick={onClickNext}>&gt;</button>

        <div className="logo">
          <img alt="â˜‘ Trustpilot" src={logoSrc} />
        </div>

        <div className="stars">
          <StarFill floating rating={averageReviewRating(reviews)} />
        </div>
      </div>
    );
  }
}

Widget.propTypes = {
  reviews: PropTypes.arrayOf(
    PropTypes.shape({
      fullName: PropTypes.string,
      location: PropTypes.string,
      reviewBody: PropTypes.string,
      reviewTitle: PropTypes.string,
      starRating: PropTypes.string,
    })
  ),
  currentCard: PropTypes.number.isRequired,
  onClickBack: PropTypes.func.isRequired,
  onClickNext: PropTypes.func.isRequired,
};

Widget.defaultProps = {
  reviews: [],
};

export default withSpinner(Widget);
