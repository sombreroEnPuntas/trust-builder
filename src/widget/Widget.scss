@import './styles';

.Widget {
  position: relative;
  width: 870px;
  margin: auto;

  .button {
    @include button-normal();
    z-index: 3;
    position: absolute;
    top: calc(50% - #{$font-size-button-normal-block-half});

    &_type-next {
      right: -10%;
    }

    &_type-back {
      left: -10%;
    }
  }

  .comments {
    @include display-flex-center();
    height: $widget-block-normal;
  
    .positioning {
      display: inline-block;
    }

    .positioning:nth-child(1) {
      z-index: 0;
      transform: rotate(-60deg) scale(0.01) translate(1500%, 18500%);
    }
    .positioning:nth-child(2) {
      z-index: 0;
      transform: rotate(-40deg) scale(0.7) translate(60%, 120%);
    }
    .positioning:nth-child(3) {
      z-index: 1;
      transform: rotate(-20deg) scale(0.9) translate(35%,25%);
    }
    .positioning:nth-child(4) {
      z-index: 2;
    }
    .positioning:nth-last-child(3) {
      z-index: 1;
      transform: rotate(20deg) scale(0.9) translate(-35%,25%);
    }
    .positioning:nth-last-child(2) {
      z-index: 0;
      transform: rotate(40deg) scale(0.7) translate(-60%, 120%);
    }
    .positioning:nth-last-child(1) {
      z-index: 0;
      transform: rotate(60deg) scale(0.01) translate(-1500%, 18500%);
    }
  }
  
  .logo {
    @include display-flex-center();
    margin-top: 20px;
  }
  
  .stars {
    margin-top: 30px;
  }
}
