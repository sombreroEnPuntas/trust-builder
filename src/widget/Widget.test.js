import React from 'react';
import { mount } from 'enzyme';

import reviews from '../mocks/reviews';
import Widget from './Widget';

it('Widget renders without crashing', () => {
  mount(<Widget totalCards={0} />);
});

describe('Widget renders mocks as expected for:', () => {
  it('Shows loading message...', () => {
    const wrapper = mount(<Widget totalCards={0} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('Shows reviews!', () => {
    const wrapper = mount(<Widget reviews={reviews} totalCards={reviews.length} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
