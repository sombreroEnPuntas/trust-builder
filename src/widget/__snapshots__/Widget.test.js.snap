// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget renders mocks as expected for: Shows loading message... 1`] = `
"<WithSpinner totalCards={0} initialCard={0}>
  <Widget totalCards={0} currentCard={0} onClickBack={[Function]} onClickNext={[Function]} reviews={{...}}>
    <div>
      Loading reviews...
    </div>
  </Widget>
</WithSpinner>"
`;

exports[`Widget renders mocks as expected for: Shows reviews! 1`] = `
"<WithSpinner reviews={{...}} totalCards={15} initialCard={0}>
  <Widget reviews={{...}} totalCards={15} currentCard={0} onClickBack={[Function]} onClickNext={[Function]}>
    <div className=\\"Widget\\">
      <button className=\\"button button_type-back\\" onClick={[Function]}>
        &lt;
      </button>
      <div className=\\"comments\\">
        <span className=\\"positioning\\">
          <Comment review={{...}} size=\\"normal\\">
            <div className=\\"Comment Comment_size-normal\\">
              <div className=\\"header\\">
                <span className=\\"title\\">
                  Simon Lock
                </span>
                <span className=\\"rating\\">
                  <StarFill rating={5} floating={false}>
                    <span className=\\"StarFill\\">
                      <Star index={1} rating={5} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-5 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                      <Star index={2} rating={5} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-5 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                      <Star index={3} rating={5} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-5 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                      <Star index={4} rating={5} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-5 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                      <Star index={5} rating={5} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-5 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                    </span>
                  </StarFill>
                </span>
                <div className=\\"sub-title\\">
                  <span>
                    Kolding
                  </span>
                </div>
              </div>
              <div className=\\"body\\">
                <p className=\\"title\\">
                  Super quality.. I will show here again!
                </p>
                <p>
                  Super nice quality, fast devilery, good prices. I will shop here again!
                </p>
              </div>
            </div>
          </Comment>
        </span>
        <span className=\\"positioning\\">
          <Comment review={{...}} size=\\"normal\\">
            <div className=\\"Comment Comment_size-normal\\">
              <div className=\\"header\\">
                <span className=\\"title\\">
                  Gav
                </span>
                <span className=\\"rating\\">
                  <StarFill rating={4} floating={false}>
                    <span className=\\"StarFill\\">
                      <Star index={1} rating={4} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-4 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                      <Star index={2} rating={4} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-4 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                      <Star index={3} rating={4} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-4 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                      <Star index={4} rating={4} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-4 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                      <Star index={5} rating={4} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-0 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                    </span>
                  </StarFill>
                </span>
                <div className=\\"sub-title\\">
                  <span />
                </div>
              </div>
              <div className=\\"body\\">
                <p className=\\"title\\">
                  Princely Sum
                </p>
                <p>
                  A decent local curry house in Faversham, Kent known for its Elvis nights.
                </p>
              </div>
            </div>
          </Comment>
        </span>
        <span className=\\"positioning\\">
          <Comment review={{...}} size=\\"normal\\">
            <div className=\\"Comment Comment_size-normal\\">
              <div className=\\"header\\">
                <span className=\\"title\\">
                  Justin Wright
                </span>
                <span className=\\"rating\\">
                  <StarFill rating={3} floating={false}>
                    <span className=\\"StarFill\\">
                      <Star index={1} rating={3} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-3 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                      <Star index={2} rating={3} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-3 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                      <Star index={3} rating={3} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-3 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                      <Star index={4} rating={3} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-0 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                      <Star index={5} rating={3} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-0 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                    </span>
                  </StarFill>
                </span>
                <div className=\\"sub-title\\">
                  <span>
                    London, GB
                  </span>
                </div>
              </div>
              <div className=\\"body\\">
                <p className=\\"title\\">
                  Good Services
                </p>
                <p>
                  A decent place to introduce your taste buds to fiery Indian fare
                </p>
              </div>
            </div>
          </Comment>
        </span>
        <span className=\\"positioning\\">
          <Comment review={{...}} size=\\"normal\\">
            <div className=\\"Comment Comment_size-normal\\">
              <div className=\\"header\\">
                <span className=\\"title\\">
                  Erika Wolfe
                </span>
                <span className=\\"rating\\">
                  <StarFill rating={2} floating={false}>
                    <span className=\\"StarFill\\">
                      <Star index={1} rating={2} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-2 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                      <Star index={2} rating={2} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-2 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                      <Star index={3} rating={2} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-0 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                      <Star index={4} rating={2} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-0 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                      <Star index={5} rating={2} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-0 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                    </span>
                  </StarFill>
                </span>
                <div className=\\"sub-title\\">
                  <span>
                    Gothenburg, SE
                  </span>
                </div>
              </div>
              <div className=\\"body\\">
                <p className=\\"title\\">
                  Nightmare experience - no product, no communication, no refund; improved by rapid resolution
                </p>
                <p>
                  In early 2012, I ordered a set of chairs from Infurn. I thought that by ordering in March, I would have what I needed by November, certainly. I wanted the perfect chairs for my house, and really did not NEED them before the annual Thanksgiving dinner (the only time of year I have a need for a whole lot of chairs at one time). I played it safe, I thought, by ordering so far in advance.
                  Week after week,
                  month after month,
                  Infurn &#39;s website kept updating the status of the order so that the chairs&#39;
                  arrival would be further and further in the distant future.Finally when the week of Thanksgiving arrived,
                  and I had had contact with their customer service in early November(and their only clueless reply was to say, &#39;We had manufacturing problems; let&#39;
                  s hope the chairs arrive this week as scheduled.
                  &#39;), I still had no chairs, had to go out and buy some other chairs and still had no clue when the Infurn chairs might arrive.
                  At some point I
                  finally just requested a refund because Infurn could neither deliver my chairs nor give me a solid date about when I might receive them when I inquired about a delivery date.They
                  finally offered me a refund - which I accepted on 14 December 2012. 
                </p>
              </div>
            </div>
          </Comment>
        </span>
        <span className=\\"positioning\\">
          <Comment review={{...}} size=\\"normal\\">
            <div className=\\"Comment Comment_size-normal\\">
              <div className=\\"header\\">
                <span className=\\"title\\">
                  Hugo Beja
                </span>
                <span className=\\"rating\\">
                  <StarFill rating={1} floating={false}>
                    <span className=\\"StarFill\\">
                      <Star index={1} rating={1} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-1 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                      <Star index={2} rating={1} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-0 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                      <Star index={3} rating={1} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-0 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                      <Star index={4} rating={1} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-0 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                      <Star index={5} rating={1} size=\\"small\\">
                        <span className=\\"Star\\">
                          <span className=\\"star-block_color-0 star-block_size-small\\">
                            ðŸŸŠ
                          </span>
                        </span>
                      </Star>
                    </span>
                  </StarFill>
                </span>
                <div className=\\"sub-title\\">
                  <span>
                    Praia Da Barra, PT
                  </span>
                </div>
              </div>
              <div className=\\"body\\">
                <p className=\\"title\\">
                  FRAUD
                </p>
                <p>
                  I&#39;ve been patiently waiting for a miracle to happen with our order AU-316084, 12 Chairs!! First the delays, lots and lots of delays with no apparent reason... after they send just 1/2 of the order and say it&#39;s all... So we bought &#39;Pairs&#39; of chairs... we paid for 5 PAIRS and received 5 chairs!!! The other 2, the rocket chairs we never receive....
                  Communication ZERO,
                  they simply ignore the e - mails...when they did respond they asked us to prove our order to be pairs!!!LOL and just stopped communicating...their website is constantly down...probably to make lose interest and rest your forces to recover what you paid
                  for !!
                </p>
              </div>
            </div>
          </Comment>
        </span>
      </div>
      <button className=\\"button button_type-next\\" onClick={[Function]}>
        &gt;
      </button>
      <div className=\\"logo\\">
        <img alt=\\"â˜‘ Trustpilot\\" src=\\"/logo/255x29.png\\" />
      </div>
      <div className=\\"stars\\">
        <StarFill floating={true} rating={3}>
          <span className=\\"StarFill-floating\\">
            <span className=\\"positioning\\">
              <Star index={1} rating={3} size=\\"big\\">
                <span className=\\"Star\\">
                  <span className=\\"star-block_color-3 star-block_size-big\\">
                    ðŸŸŠ
                  </span>
                </span>
              </Star>
            </span>
            <span className=\\"positioning\\">
              <Star index={2} rating={3} size=\\"big\\">
                <span className=\\"Star\\">
                  <span className=\\"star-block_color-3 star-block_size-big\\">
                    ðŸŸŠ
                  </span>
                </span>
              </Star>
            </span>
            <span className=\\"positioning\\">
              <Star index={3} rating={3} size=\\"big\\">
                <span className=\\"Star\\">
                  <span className=\\"star-block_color-3 star-block_size-big\\">
                    ðŸŸŠ
                  </span>
                </span>
              </Star>
            </span>
            <span className=\\"positioning\\">
              <Star index={4} rating={3} size=\\"big\\">
                <span className=\\"Star\\">
                  <span className=\\"star-block_color-0 star-block_size-big\\">
                    ðŸŸŠ
                  </span>
                </span>
              </Star>
            </span>
            <span className=\\"positioning\\">
              <Star index={5} rating={3} size=\\"big\\">
                <span className=\\"Star\\">
                  <span className=\\"star-block_color-0 star-block_size-big\\">
                    ðŸŸŠ
                  </span>
                </span>
              </Star>
            </span>
          </span>
        </StarFill>
      </div>
    </div>
  </Widget>
</WithSpinner>"
`;
